sdk/parents/clientcore-parent/pom.xml

```diff
@@ -733,10 +731,19 @@
               <arg>-Xlint:-requires-transitive-automatic</arg> <!-- FIXME: this is required for now as it introduces a build failure -->
             </compilerArgs>
 
-            <!-- Default is annotation processing being off. -->
-            <proc>${compiler.proc}</proc>
-
+            <!-- Default is annotation processing being on. -->
+            <annotationProcessorPaths>
```

Comment:
We might want to have this behind a flag so only libraries that need annotation processing run it. We likely don't want this to be ran on the ClientCore Core libraries other than the special testing package.

Reply:
Added a flag to enable this only for the e2e module testing `annotation-processor-tests` https://github.com/Azure/azure-sdk-for-java/pull/43886/commits/8aac751e612a311b00c27db13fe94e77cc4dbd75